File: source/resources/examples/javascript/namespace/RegisteringASubnamespace.js
/*
 *
 * Copyright 2018 NEM
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

const nem2Sdk = require("nem2-sdk");
const Account = nem2Sdk.Account,
    Deadline = nem2Sdk.Deadline,
    NetworkType = nem2Sdk.NetworkType,
    RegisterNamespaceTransaction = nem2Sdk.RegisterNamespaceTransaction,
    TransactionHttp = nem2Sdk.TransactionHttp;

// Replace with private key
const privateKey = process.env.PRIVATE_KEY;

const account = Account.createFromPrivateKey(privateKey, NetworkType.MIJIN_TEST);

//Replace with root namespace name
const rootNamespaceName = 'foo';

//Replace with subnamespace name
const subnamespace = 'bar';

const registerNamespaceTransaction = RegisterNamespaceTransaction.createSubNamespace(
    Deadline.create(),
    subnamespace,
    rootNamespaceName,
    NetworkType.MIJIN_TEST
);

const signedTransaction = account.sign(registerNamespaceTransaction);

const transactionHttp = new TransactionHttp('http://localhost:3000');

transactionHttp.announce(signedTransaction).subscribe(x => console.log(x));

File: source/resources/examples/java/src/test/java/nem2/guides/examples/namespace/RegisteringASubnamespace.java
/*
 *
 * Copyright 2018-present NEM
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package nem2.guides.examples.namespace;

import io.nem.sdk.api.RepositoryFactory;
import io.nem.sdk.api.TransactionRepository;
import io.nem.sdk.infrastructure.vertx.RepositoryFactoryVertxImpl;
import io.nem.sdk.model.account.Account;
import io.nem.sdk.model.blockchain.NetworkType;
import io.nem.sdk.model.namespace.NamespaceId;
import io.nem.sdk.model.transaction.NamespaceRegistrationTransaction;
import io.nem.sdk.model.transaction.NamespaceRegistrationTransactionFactory;
import io.nem.sdk.model.transaction.SignedTransaction;
import java.util.concurrent.ExecutionException;
import org.junit.jupiter.api.Test;

class RegisteringASubnamespace {

    @Test
    void registeringASubnamespace()
        throws ExecutionException, InterruptedException {

        try (final RepositoryFactory repositoryFactory = new RepositoryFactoryVertxImpl(
            "http://localhost:3000")) {
            final String generationHash = repositoryFactory.getGenerationHash().toFuture().get();

            final NetworkType networkType = repositoryFactory
                .getNetworkType().toFuture().get();

            final TransactionRepository transactionRepository = repositoryFactory
                .createTransactionRepository();
            /* start block 01 */
            // Replace with private key
            final String privateKey = "";

            final Account account = Account
                .createFromPrivateKey(privateKey, networkType);

            // Replace with root namespace name
            final NamespaceId rootNamespaceId = new NamespaceId("foo");

            //Replace with subnamespace name
            final String subnamespaceName = "bar";

            final NamespaceRegistrationTransaction registerNamespaceTransaction = NamespaceRegistrationTransactionFactory
                .createSubNamespace(
                    networkType,
                    subnamespaceName,
                    rootNamespaceId
                ).build();

            final SignedTransaction signedTransaction = account
                .sign(registerNamespaceTransaction, generationHash);

            transactionRepository.announce(signedTransaction).toFuture().get();
            /* end block 01 */
        }
    }
}
